src/colmap/scene/database.cc

```diff
@@ -217,50 +217,66 @@ std::optional<std::stringstream> BlobColumnToStringStream(
   return stream;
 }
 
-Rig ReadRigRow(sqlite3_stmt* sql_stmt) {
+void ReadRigRows(sqlite3_stmt* sql_stmt,
+                 const std::function<void(Rig)>& new_rig_callback) {
   Rig rig;
+  while (SQLITE3_CALL(sqlite3_step(sql_stmt)) == SQLITE_ROW) {
+    const rig_t rig_id = static_cast<rig_t>(sqlite3_column_int64(sql_stmt, 0));
+    if (rig_id != rig.RigId()) {
```

Comment:
Here we are assuming that all rows for a particular rig will be consecutive. I'm not sure that an SQL JOIN guaranties this. So it is likely that an `ORDER BY rigs.rig_id` is needed in the query below.